doctest AFNIHeader.get_data_scaling fails if ran "alone"

originally reported in https://github.com/nipy/nibabel/pull/782 but has nothing to do with the endianess, so filing separately

when I ran the full sweep using `python -m nose -s -v --with-doctest  nibabel 2>&1 | tee tests.log` that doctest was reported as passed:
```
(sid_s390x-dchroot)yoh@zelenka:~/nibabel$ grep AFNIHeader.get_data_scaling  tests.log 
Doctest: nibabel.brikhead.AFNIHeader.get_data_scaling ... ok
``` 
but when I run only the doctest - it fails
```shell
(sid_s390x-dchroot)yoh@zelenka:~/nibabel$ python -m nose -s -v --with-doctest  nibabel/brikhead.py
Doctest: nibabel.brikhead.AFNIHeader.__init__ ... ok
Doctest: nibabel.brikhead.AFNIHeader.get_affine ... ok
Doctest: nibabel.brikhead.AFNIHeader.get_data_scaling ... FAIL
Doctest: nibabel.brikhead.AFNIHeader.get_volume_labels ... ok
Doctest: nibabel.brikhead.AFNIImage ... ok
Doctest: nibabel.brikhead._unpack_var ... ok
Doctest: nibabel.brikhead.parse_AFNI_header ... ok

======================================================================
FAIL: Doctest: nibabel.brikhead.AFNIHeader.get_data_scaling
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/doctest.py", line 2224, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for nibabel.brikhead.AFNIHeader.get_data_scaling
  File "/home/yoh/nibabel/nibabel/brikhead.py", line 414, in get_data_scaling

----------------------------------------------------------------------
File "/home/yoh/nibabel/nibabel/brikhead.py", line 422, in nibabel.brikhead.AFNIHeader.get_data_scaling
Failed example:
    header.get_data_scaling()
Expected:
    array([  3.88336300e-08])
Got:
    array([3.883363e-08])


----------------------------------------------------------------------
Ran 7 tests in 0.011s
```
that is numpy 1.16.2 .  Happens to me also on my laptop (regular little endian amd64) with the same numpy
```
(git)hopa:~nipy-suite/nibabel[master]git
$> python -m nose -s -v --with-doctest  nibabel/brikhead.py
Doctest: nibabel.brikhead.AFNIHeader.__init__ ... ok
Doctest: nibabel.brikhead.AFNIHeader.get_affine ... ok
Doctest: nibabel.brikhead.AFNIHeader.get_data_scaling ... FAIL
Doctest: nibabel.brikhead.AFNIHeader.get_volume_labels ... ok
Doctest: nibabel.brikhead.AFNIImage ... ok
Doctest: nibabel.brikhead._unpack_var ... ok
Doctest: nibabel.brikhead.parse_AFNI_header ... ok

======================================================================
FAIL: Doctest: nibabel.brikhead.AFNIHeader.get_data_scaling
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/doctest.py", line 2224, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for nibabel.brikhead.AFNIHeader.get_data_scaling
  File "/home/yoh/proj/nipy/nipy-suite/nibabel/nibabel/brikhead.py", line 414, in get_data_scaling

----------------------------------------------------------------------
File "/home/yoh/proj/nipy/nipy-suite/nibabel/nibabel/brikhead.py", line 422, in nibabel.brikhead.AFNIHeader.get_data_scaling
Failed example:
    header.get_data_scaling()
Expected:
    array([  3.88336300e-08])
Got:
    array([3.883363e-08])


----------------------------------------------------------------------
Ran 7 tests in 0.017s

FAILED (failures=1)
```

--- a/nibabel/__init__.py
+++ b/nibabel/__init__.py
@@ -36,27 +36,27 @@ For more detailed information see the :ref:`manual`.
 """
 
 # Package-wide test setup and teardown
-# Numpy changed print options in 1.14; we can update docstrings and remove
-# these when our minimum for building docs exceeds that
-_save_printopts = None
+_test_states = {
+    # Numpy changed print options in 1.14; we can update docstrings and remove
+    # these when our minimum for building docs exceeds that
+    'legacy_printopt': None,
+    }
 
 def setup_package():
     """ Set numpy print style to legacy="1.13" for newer versions of numpy """
-    import nibabel as nb
     import numpy as np
     from distutils.version import LooseVersion
-    if nb._save_printopts is None:
-        nb._save_printopts = np.get_printoptions().get('legacy')
     if LooseVersion(np.__version__) >= LooseVersion('1.14'):
+        if _test_states.get('legacy_printopt') is None:
+            _test_states['legacy_printopt'] = np.get_printoptions().get('legacy')
         np.set_printoptions(legacy="1.13")
 
 def teardown_package():
     """ Reset print options when tests finish """
-    import nibabel as nb
     import numpy as np
-    if nb._save_printopts is not None:
-        np.set_printoptions(legacy=nb._save_printopts)
-        nb._save_printopts = None
+    if _test_states.get('legacy_printopt') is not None:
+        np.set_printoptions(legacy=_test_states['legacy_printopt'])
+        _test_states['legacy_printopt'] = None
 
 
 # module imports

--- a/nibabel/__init__.py
+++ b/nibabel/__init__.py
@@ -35,16 +35,28 @@ Quickstart
 For more detailed information see the :ref:`manual`.
 """
 
+# Package-wide test setup and teardown
+# Numpy changed print options in 1.14; we can update docstrings and remove
+# these when our minimum for building docs exceeds that
+_save_printopts = None
 
-def setup_test():
-    """ Set numpy print options to "legacy" for new versions of numpy
-
-    If imported into a file, nosetest will run this before any doctests.
-    """
-    import numpy
+def setup_package():
+    """ Set numpy print style to legacy="1.13" for newer versions of numpy """
+    import nibabel as nb
+    import numpy as np
     from distutils.version import LooseVersion
-    if LooseVersion(numpy.__version__) >= LooseVersion('1.14'):
-        numpy.set_printoptions(legacy="1.13")
+    if nb._save_printopts is None:
+        nb._save_printopts = np.get_printoptions().get('legacy')
+    if LooseVersion(np.__version__) >= LooseVersion('1.14'):
+        np.set_printoptions(legacy="1.13")
+
+def teardown_package():
+    """ Reset print options when tests finish """
+    import nibabel as nb
+    import numpy as np
+    if nb._save_printopts is not None:
+        np.set_printoptions(legacy=nb._save_printopts)
+        nb._save_printopts = None
 
 
 # module imports

--- a/nibabel/__init__.py
+++ b/nibabel/__init__.py
@@ -55,8 +55,7 @@ def teardown_package():
     """ Reset print options when tests finish """
     import numpy as np
     if _test_states.get('legacy_printopt') is not None:
-        np.set_printoptions(legacy=_test_states['legacy_printopt'])
-        _test_states['legacy_printopt'] = None
+        np.set_printoptions(legacy=_test_states.pop('legacy_printopt'))
 
 
 # module imports

--- a/nibabel/affines.py
+++ b/nibabel/affines.py
@@ -6,7 +6,6 @@
 import numpy as np
 
 from six.moves import reduce
-from . import setup_test  # noqa
 
 
 class AffineError(ValueError):

--- a/nibabel/casting.py
+++ b/nibabel/casting.py
@@ -8,7 +8,6 @@ from numbers import Integral
 from platform import processor, machine
 
 import numpy as np
-from . import setup_test  # noqa
 
 
 class CastingError(Exception):

--- a/nibabel/nicom/dwiparams.py
+++ b/nibabel/nicom/dwiparams.py
@@ -21,7 +21,6 @@ The B matrix ``B`` is a symmetric positive semi-definite matrix.  If
 '''
 import numpy as np
 import numpy.linalg as npl
-from .. import setup_test as setup_module  # noqa
 
 
 def B2q(B, tol=None):

--- a/nibabel/nifti1.py
+++ b/nibabel/nifti1.py
@@ -28,7 +28,6 @@ from . import analyze  # module import
 from .spm99analyze import SpmAnalyzeHeader
 from .casting import have_binary128
 from .pydicom_compat import have_dicom, pydicom as pdcm
-from . import setup_test  # noqa
 
 # nifti1 flat header definition for Analyze-like first 348 bytes
 # first number in comments indicates offset in file header in bytes

--- a/nibabel/quaternions.py
+++ b/nibabel/quaternions.py
@@ -27,7 +27,6 @@ they are applied on the left of the vector.  For example:
 
 import math
 import numpy as np
-from . import setup_test  # noqa
 
 MAX_FLOAT = np.maximum_sctype(np.float)
 FLOAT_EPS = np.finfo(np.float).eps

